File: app/src/main/java/com/antisahir/app/AdminReceiver.java
package com.antisahir.app;

import android.app.admin.DeviceAdminReceiver;
import android.content.Context;
import android.content.Intent;
import android.widget.Toast;

public class AdminReceiver extends DeviceAdminReceiver {

    @Override
    public void onEnabled(Context context, Intent intent) {
        Toast.makeText(context,
                "مدير الجهاز مُفعَّل ✅ — لا يمكن حذف التطبيق قبل انتهاء البرنامج",
                Toast.LENGTH_LONG).show();
    }

    @Override
    public CharSequence onDisableRequested(Context context, Intent intent) {
        return "⚠️ تعطيل مدير الجهاز سيسمح بحذف التطبيق. هل أنت متأكد؟";
    }

    @Override
    public void onDisabled(Context context, Intent intent) {
        Toast.makeText(context,
                "تم تعطيل مدير الجهاز",
                Toast.LENGTH_SHORT).show();
    }
}

File: app/AdminReceiver.kt
package com.focusguard.app

import android.app.admin.DeviceAdminReceiver
import android.content.Context
import android.content.Intent
import android.widget.Toast
import java.util.Calendar

class AdminReceiver : DeviceAdminReceiver() {

    override fun onEnabled(context: Context, intent: Intent) {
        Toast.makeText(context, "✓ صلاحيات المشرف مُفعَّلة", Toast.LENGTH_SHORT).show()
    }

    override fun onDisableRequested(context: Context, intent: Intent): CharSequence {
        val prefs = PrefsManager(context)
        return if (prefs.isScheduleActive && isBlocking(prefs))
            "⚠ لا يمكن التعطيل أثناء فترة حجب نشطة!"
        else
            "هل تريد تعطيل حارس التركيز؟"
    }

    override fun onDisabled(context: Context, intent: Intent) {
        Toast.makeText(context, "تم تعطيل صلاحيات المشرف", Toast.LENGTH_SHORT).show()
    }

    private fun isBlocking(prefs: PrefsManager): Boolean {
        val now = Calendar.getInstance()
        val cur = now.get(Calendar.HOUR_OF_DAY) * 60 + now.get(Calendar.MINUTE)
        val s = prefs.startHour * 60 + prefs.startMinute
        val e = prefs.endHour * 60 + prefs.endMinute
        return if (s <= e) cur in s until e else cur >= s || cur < e
    }
}
